src/GraphQLParser/AST/GraphQLComment.cs

```diff
@@ -17,17 +16,22 @@ public class GraphQLComment : ASTNode
         /// </summary>
         public ROM Text { get; set; }
 
-        public override GraphQLLocation Location
-        {
-            get => _location;
-            set => _location = value;
-        }
-
         // Comment itself can't have a comment
         //public override GraphQLComment? Comment
         //{
         //    get => _comment;
         //    set => _comment = value;
         //}
     }
+
+    internal class GraphQLCommentFull : GraphQLComment
+    {
+        private GraphQLLocation _location;
+
+        public override GraphQLLocation Location
+        {
+            get => _location;
+            set => _location = value;
+        }
+    }
```

Comment:
Nowhere else do we have a special class for comments without a location.  Optimization is only for ignoring both comments and locations.

Reply:
I renamed `GraphQLCommentFull` into `GraphQLCommentWithLocation`. I will soon make a commit, where I added a lot of new internal classes to align the behavior. All optimizations will be in place. I will need some time to finish and write tests. I want to show in advance what it turns out.